httpx/client.py

```diff
@@ -1126,6 +1216,43 @@ def dispatcher_for_url(self, url: URL) -> AsyncDispatcher:
 
         return response
 
+    async def send_handling_retries(
+        self,
+        request: Request,
+        auth: Auth,
+        retries: Retries,
+        timeout: Timeout,
+        allow_redirects: bool = True,
+    ) -> Response:
+        backend = lookup_backend()
+
+        retries_left = retries.limit
+        delays = retries.get_delays()
+
+        while True:
+            try:
+                return await self.send_handling_redirects(
+                    request,
+                    auth=auth,
+                    timeout=timeout,
+                    allow_redirects=allow_redirects,
+                )
+            except HTTPError as exc:
+                if not retries.limit or not retries.should_retry_on_exception(exc):
```

Comment:
Should that be `if not retries_left`?

Reply:
Yes, this ties into your comment on skipping the retries stuff altogether when retries are disabled. Addressed in the latest commit by not entering this loop at all when that's the case.